### ISSUE TITLE
pyvista freeze 

### ISSUE THREAD

**Issue by fghoussen** [pyvista/pyvista#4462]
### Describe the bug, what's wrong, and what you expected.

pyvista freezes / hangs when used with docker.
expectation: any help, clue or workaround on the problem.

### Steps to reproduce the bug.

All is described here: https://github.com/jorgensd/dolfinx-tutorial/issues/130#issuecomment-1565458011

Note: I've just tried with dolfinx:latest docker image: exact same result

### System Information

```shell
>> sudo docker run --rm -it -e DISPLAY="$DISPLAY" -v /tmp/.X11-unix:/tmp/.X11-unix -v /tmp:/root --name fenics_tuto dolfinx/dolfinx:latest
root@3858c2c6d32d:... # Run here exact same command than in https://github.com/jorgensd/dolfinx-tutorial/issues/130
root@3858c2c6d32d:~/dolfinx-tutorial/chapter2# python3 
Python 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyvista as pv
>>> print(pv.Report())

--------------------------------------------------------------------------------
  Date: Sun May 28 10:11:37 2023 UTC

                OS : Linux
            CPU(s) : 2
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : Mesa/X.org
      GPU Renderer : llvmpipe (LLVM 13.0.1, 256 bits)
       GPU Version : 4.5 (Core Profile) Mesa 22.0.5
  MathText Support : False

  Python 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0]

           pyvista : 0.39.1
               vtk : 9.2.6
             numpy : 1.23.3
        matplotlib : 3.6.2
            scooby : 0.7.2
             pooch : v1.7.0
           IPython : 8.6.0
             scipy : 1.9.3
--------------------------------------------------------------------------------
>>>
```

**Comment by fghoussen** [comment]
Note: https://github.com/pyvista/pyvista-support/issues/128 doesn't help

**Comment by fghoussen** [comment]
Solution (or workaround ?) found: removing `pyvista.start_xvfb()`.
Is this a bug or an expected behavior, I let you tell me! I let the issue open in case PR is needed.

**Comment by banesullivan** [comment]
Could you try installing VTK with one of VTK's official OSMesa wheel variants? You can find it here for the 9.2.6 release:

https://gitlab.kitware.com/vtk/vtk/-/packages/102

Please not that `pip` will see this as a different package from `vtk` so you need to uninstall `vtk` to make sure there isn't a conflict

```
pip uninstall --yes vtk
pip install vtk-osmesa --index-url https://gitlab.kitware.com/api/v4/projects/13/packages/pypi/simple
```


Some more details https://discourse.vtk.org/t/status-update-vtk-python-wheels/11212

Hope this helps

**Comment by fghoussen** [comment]
Unfortunately doesn't work: there is no GUI even in the `demo_poisson` case (which was working)

```
root@e85b3154aa8a:~# pip uninstall --yes vtk
root@e85b3154aa8a:~# pip install vtk-osmesa --index-url https://gitlab.kitware.com/api/v4/projects/13/packages/pypi/simple
root@e85b3154aa8a:~# python3 demo_poisson.py
root@e85b3154aa8a:~/dolfinx-tutorial/chapter2# python3 linearelasticity_code.py 
root@e85b3154aa8a:~/dolfinx-tutorial/chapter2#
```

**Comment by banesullivan** [comment]
Well if you're running in docker, there won't be a GUI due to the docker container being headless.

I'm a bit confused now on what your exact issue is. Would you please clarify?

**Comment by fghoussen** [comment]
The issue is about export display from docker container to host PC.

Here https://github.com/jorgensd/dolfinx-tutorial/issues/130#issuecomment-1565458011, in the first example `demo_poisson.py`, the display is exported from the docker container to the host PC : at PC side, the GUI appears and you can navigate  through the mesh. But, in the second example, `linearelasticity_code.py` doesn't show any GUI.

Export display works in some case but not in all cases.

**Comment by banesullivan** [comment]
I don't do this sort of export display often enough to have any quick tips or solutions. I suspect this isn't a bug in PyVista but something going awry in the docker configurations and graphics forwarding but I totally don't know. Could you try to put together a minimal example for us?

**Comment by fghoussen** [comment]
> Could you try to put together a minimal example for us?

No, unfortunatelly. I was just trying to run tutorial examples of FEniCS-dolphin https://github.com/jorgensd/dolfinx-tutorial/issues/130#issuecomment-1565458011.
